webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java

```diff
@@ -50,18 +54,29 @@ public StudentGroupBatch upload(final User user, final MultipartFile file) {
 
         final StudentGroupBatch.Builder studentGroupBatchBuilder = StudentGroupBatch.builder().creator(user.getUsername());
 
-        try {
-            final String digest = DigestUtils.md5Hex(validate(file.getBytes())).toUpperCase();
+        final MessageDigest messageDigest = DigestUtils.getMd5Digest();
+        try (final DigestInputStream digestStream = new DigestInputStream(file.getInputStream(), messageDigest)) {
+            validate(digestStream);
```

Comment:
Changed this to use a DigestInputStream and validate the CSV against an InputStream as well, so we never have to load the payload fully into memory.